src/data/moviesSlice.js

```diff
@@ -0,0 +1,27 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
+
+export const fetchMovies = createAsyncThunk('fetch-movies', async (apiUrl) => {
+    const response = await fetch(apiUrl)
+    return response.json()
+})
+
+const moviesSlice = createSlice({
+    name: 'movies',
+    initialState: { 
+        movies: [],
+        fetchStatus: '',
+    },
+    reducers: {},
+    extraReducers: (builder) => {
```

Comment:
I was thinking about leveraging **RTK Query**, which simplifies the entire process by handling data fetching, state management, and even caching seamlessly. By adopting RTK Query, it would eliminate the need for additional reducers and selectors.

RTK Query simplifies the handling of fetching movies and automatically manages the loading, success, and error states without explicitly defining reducers. Plus, caching is set up by default, requiring no extra configuration.

I believe incorporating RTK Query could enhance the efficiency and maintainability of this codebase. Of course, feel free to discuss or make any adjustments as per your preference. I'd be glad to assist further if needed.